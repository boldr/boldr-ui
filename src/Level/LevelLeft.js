/* @flow */
import React from 'react';
import classNames from 'classnames';
import { StyleClasses } from '../theme/styleClasses';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  className?: string,
};

const BASE_ELEMENT = StyleClasses.LEVEL_LEFT;

export function LevelLeft({ tag = 'div', ...props }: Props) {
  const className = classNames(BASE_ELEMENT, props.className);

  return React.createElement(tag, { ...props, className });
}

export default createWrappedComponent(LevelLeft);
