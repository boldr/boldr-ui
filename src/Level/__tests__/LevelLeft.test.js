import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { LevelLeft } from '../LevelLeft';

describe('<LevelLeft />', () => {
  it('should render a div with .boldrui-level__left', () => {
    const component = shallow(<LevelLeft>My LevelLeft</LevelLeft>);
    expect(component.contains(<div className="boldrui-level__left">My LevelLeft</div>)).toBe(true);
  });

  it('should render an aside with .boldrui-level__left', () => {
    const component = shallow(<LevelLeft tag="aside">My LevelLeft</LevelLeft>);
    expect(component.contains(<aside className="boldrui-level__left">My LevelLeft</aside>)).toBe(
      true,
    );
  });

  it('should render a div with .boldrui-level__left and custom classNames', () => {
    const component = shallow(
      <LevelLeft className="custom">
        <span>Any Content</span>
      </LevelLeft>,
    );
    expect(component.hasClass('boldrui-level__left')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<LevelLeft>My LevelLeft</LevelLeft>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
