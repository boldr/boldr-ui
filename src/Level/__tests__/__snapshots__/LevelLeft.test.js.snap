// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelLeft /> should match the snapshot 1`] = `
<div
  className="boldrui-level__left"
>
  My LevelLeft
</div>
`;
